### ISSUE TITLE
Empty S3Exception

### ISSUE THREAD

**Issue by francisdb** [akka/alpakka#1589]
Alpakka version: 1.0-M3

### Expected Behavior

If S3 responds with an empty response (because of empty string key/secret). I would expect a useful error message but you get an empty S3Exception

### Actual Behavior

```
Caused by: akka.stream.alpakka.s3.S3Exception:
	at akka.stream.alpakka.s3.impl.S3Stream$.$anonfun$getObjectMetadata$7(S3Stream.scala:174)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	at akka.http.scaladsl.util.FastFuture$FulfilledFuture.transform(FastFuture.scala:84)
	at scala.concurrent.Future.map(Future.scala:292)
	at scala.concurrent.Future.map$(Future.scala:292)
	at akka.http.scaladsl.util.FastFuture$FulfilledFuture.map(FastFuture.scala:77)
	at akka.stream.alpakka.s3.impl.S3Stream$.$anonfun$getObjectMetadata$4(S3Stream.scala:173)
	at akka.stream.impl.fusing.Map$$anon$1.onPush(Ops.scala:52)
	at akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:515)
	at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:478)
	at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:374)
	at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:588)
	at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:474)
	at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:563)
	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:730)
	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$shortCircuitBatch(ActorGraphInterpreter.scala:720)
	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:746)
```

### Related tickets

There are related issues:

https://github.com/akka/alpakka/issues/731 (xml parse of empty body)
https://github.com/akka/alpakka/issues/731 (similar with wrong fix?)

### Reproducible Test Case

Just provide an empty key ans secret to the S3Settings and try `getObjectMetadata`

Preferably it would check/show the http response status if it's not 200 / if it encounters an empty body so that the user at least has a clue what the source of this error might be.

**Comment by ennru** [comment]
Thank you for reporting, it would be perfect if you could prepare a PR to fix it.

**Comment by 2m** [comment]
Fixed in #1748
